users/views.py

```diff
@@ -43,28 +46,39 @@ class LoginView(APIView):
             '202': set_example(responses.login_202),
             '400': set_example(responses.login_400),
             '401': set_example(responses.login_401),
+            '404': set_example(responses.login_404)
         },
     )
+
+
+
     def post(self, request):
         serializer = LoginSerializer(data=request.data)
 
+        found_email = request.data['email']
```

Comment:
Good effort  🔥 👍 , as mentioned by @sahil9001 , the test cases are failing. 

We cannot be using our email without being sure that an email has been entered, so it is important to check if the serializer is valid first.
You can run tests with `python3 manage.py test` to observe that the error comes in the test case with no input, an entry our serializer would deem invalid. Also an example of how testing a variety of test cases is important.
You may shift this to the next if block where it checks for valid entries in the serializer.

It would also be nice to add a test case in `tests_login_api.py` file checking the case for absent emails.

Reply:
okay ma'am

Reply:
Thank you, ma'am, your suggestion really works